src/main/java/core/basesyntax/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportGenerator;
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private final Storage storage;
+
+    public ReportGeneratorImpl(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public String getReport(Storage storage) {
+        StringBuilder builder = new StringBuilder("fruit,quantity" + System.lineSeparator());
+        for (Map.Entry<String, Integer> entry : storage.getStorage().entrySet()) {

```

Comment:
Direct access to the `storage` map violates encapsulation. Consider adding a method in the `Storage` class to get the entries in a more encapsulated way.